UdpVideo/UdpVideoController.cpp

```diff
@@ -0,0 +1,169 @@
+        {
+            auto actualPath = QCoreApplication::applicationDirPath().replace('/', '\\').toStdWString();
+            AddDllDirectory((actualPath + L"\\bin\\").c_str());
+            AddDllDirectory((actualPath + L"\\lib\\gstreamer-1.0\\").c_str());
+
+            loadLibrary("libintl-8.dll");
+            loadLibrary("ADVAPI32.dll");
+            loadLibrary("dnsapi.dll");
+            loadLibrary("IPHLPAPI.DLL");
+            loadLibrary("KERNEL32.dll");
+            loadLibrary("msvcrt.dll");
+            loadLibrary("ole32.dll");
+            loadLibrary("SHELL32.dll");
+            loadLibrary("SHLWAPI.dll");
+            loadLibrary("USER32.dll");
+            loadLibrary("WS2_32.dll");
+            loadLibrary("zlib1.dll");
+            loadLibrary("libglib-2.0-0.dll");
+            loadLibrary("libgmodule-2.0-0.dll");
+            loadLibrary("libgobject-2.0-0.dll");
+            loadLibrary("libgstvideotestsrc.dll");
+            loadLibrary("libgstcoreelements.dll");
+            loadLibrary("libglib-2.0-0.dll");
+            loadLibrary("libgstbase-1.0-0.dll");
+            loadLibrary("liborc-0.4-0.dll");
+            loadLibrary("libgstvideo-1.0-0.dll");
+            loadLibrary("libgstx264.dll");
+            loadLibrary("libgstvideoparsersbad.dll");
+            loadLibrary("libgstrtp.dll");
+            loadLibrary("libgstudp.dll");
+
+            gstreamerHandle = loadLibrary("libgstreamer-1.0-0.dll");
+
+            if(gstreamerHandle != NULL)
+            {
+#define RESOLVE(name) name = reinterpret_cast<decltype(name)>(GetProcAddress(gstreamerHandle, #name))
+                RESOLVE(gst_init);
+                RESOLVE(gst_parse_launch);
+                RESOLVE(gst_element_set_state);
+                RESOLVE(gst_object_unref);
+#undef RESOLVE
+                gst_init(nullptr, nullptr);
+            }
+        }
+    }
+    void startPipeline(const QString& beginning, const QString& ip, const QString& port)
+    {
+        auto pipeline = beginning +
+                        " is-live=true do-timestamp=true "
+                        "! autovideoconvert "
+                        "! queue leaky=downstream "
+                        "! x264enc tune=zerolatency "
+                        "! h264parse "
+                        "! rtph264pay pt=96 "
+                        "! udpsink host=" + ip + " port=" + port;
+        gstPipeline = gst_parse_launch(pipeline.toStdString().c_str(), nullptr);
+        gst_element_set_state(gstPipeline, GST_STATE_PLAYING);
+    }
+
+    void stop()
+    {
+        if(gstPipeline != nullptr)
+        {
+            gst_element_set_state(gstPipeline, GST_STATE_NULL);
+            gst_object_unref(gstPipeline);
+            gstPipeline = nullptr;
+        }
+    }
+
+    bool hasGstreamer()
+    {
+        return gstreamerHandle != NULL;
+    }
+    HMODULE WINAPI gstreamerHandle = NULL;
+    GstElement* gstPipeline = nullptr;
+    HMODULE loadLibrary(const char* dllName)

```

Comment:
I would consider creating some abstraction for this load/unload libraries with raii and also resolve. In general for now all the load lib stuff is mixed with parsing and launching the pipeline.

Reply:
splitted